packages/theme/src/components/select.ts

```diff
@@ -1,70 +1,135 @@
 import { selectAnatomy as parts } from "@chakra-ui/anatomy"
-import type {
-  PartsStyleFunction,
-  PartsStyleObject,
-  SystemStyleFunction,
-  SystemStyleObject,
-} from "@chakra-ui/theme-tools"
-import { mergeWith } from "@chakra-ui/utils"
-import { mode } from "@chakra-ui/theme-tools"
-import Input from "./input"
+import { getColor, mode, SystemStyleFunction } from "@chakra-ui/theme-tools"
+import { Input } from "."
+
+const baseStyle: SystemStyleFunction = (props) => {
+  const { isOpen, colorScheme, invalid, theme } = props
+  const focusBorder = {
+    borderColor: mode("blue.500", "blue.300")(props),
+    boxShadow: `0 0 0 1px ${getColor(
+      theme,
+      mode("blue.500", "blue.300")(props),
+    )}`,
+  }
 
-const baseStyleField: SystemStyleFunction = (props) => {
   return {
-    ...Input.baseStyle.field,
-    bg: mode("white", "gray.700")(props),
-    appearance: "none",
-    paddingBottom: "1px",
-    lineHeight: "normal",
-    "> option, > optgroup": {
-      bg: mode("white", "gray.700")(props),
+    wrapper: {
+      position: "relative",
+    },
+    input: {
+      display: "none",
+      width: "100%",
+    },
+    trigger: {
+      ...Input.baseStyle.field,
+      justifyContent: "space-between",
+      alignItems: "center",
+      p: "12px 20px",
+      h: 0,
+      display: "flex",
+      fontSize: "sm",
+      width: "100%",
+      borderRadius: "base",
+      borderWidth: "1px",
+      borderStyle: "solid",
+      height: "100%",
+      _disabled: {
+        opacity: 0.4,
+        cursor: "not-allowed",
+      },
+      ...(invalid && {
+        borderColor: mode("red.500", "red.300")(props),
+        boxShadow: `0 0 0 1px ${getColor(
```

Comment:
Is there an alternative not to use deprecated getColor?

Reply:
@MasterGordon maybe you looked at this already, but you can leave out the color value and it'll default to `currentColor`, hence you can set the `color` property directly on the selector as a theme token (`color: "red.500"`).